crates/bevy_tasks/src/task_pool.rs

```diff
@@ -589,7 +596,10 @@ impl<'scope, 'env, T: Send + 'scope> Scope<'scope, 'env, T> {
     ///
     /// For more information, see [`TaskPool::scope`].
     pub fn spawn<Fut: Future<Output = T> + 'scope + Send>(&self, f: Fut) {
-        let task = self.executor.spawn(f).fallible();
+        let task = self
+            .executor
+            .spawn(AssertUnwindSafe(f).catch_unwind())
```

Comment:
I don't think we can guarantee that the passed in future is UnwindSafe.

Reply:
> Send + 'static pretty much implies unwind safe.

In our case the futures aren't `'static`.

Reply:
Right, so I guess the thing to do would be make it so the spawn functions can only take unwind safe futures and then assure that safety from bevy_ecs.

Reply:
Actually, I took a more in depth look at this and I think unwind safety doesn't matter here. Sure, we are catching the unwinds, but we aren't restarting tasks or returning the result, instead the panic is being propagated instead of creating a new panic. The real problem is in bevy_ecs, where catching an unwind previously led to a panic, now still runs the systems. In order to guarantee that broken invariants aren't observed, the mutable accesses of the panicked system need to be marked, and any system that depends on any of the marked accesses need to be skipped.